signcrypt_seal.go

```diff
@@ -69,6 +69,7 @@ func (sss *signcryptSealStream) signcryptBytes(b []byte) error {
 
 	nonce := nonceForChunkSigncryption(sss.numBlocks)
 
+	// TODO: Encode payload chunk.
```

Comment:
Hmm, you're totally right, I didn't think that through. Putting that flag in the hash won't save us though, because the hash is only used in the signature input.

The easiest fix I can think of is including the last packet bit in the encryption nonce somewhere, _in addition_ to putting it in the signature input. That's the closest thing NaCl gives us to associated data, unless we want to add yet another extra key derivation step. The resulting story feels a little scatterbrained to me though: "The signature covers some context, the header hash, the complete packet nonce, the plaintext hash, and the final packet flag. The outer MAC covers the ciphertext, the nonce (implicitly), and the final packet flag. It doesn't include the header hash, though."

Maybe we should have a more unified story for the associated data in this scheme. Here "the associated data" would be the header hash plus the final packet flag. Signing would cover the plaintext, the nonce, and "the associated data". Encryption would mix "the associated data" with the encryption key. Less scatterbrained?